### PR TITLE
Lint on GitHub Actions via pre-commit

### PR DESCRIPTION
**Pull request by hugovk**
Move the existing lint checks to pre-commit and add some more.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
+extend-ignore = C408,E203,F841,W503
+max-complexity=10
```
Comment (ezio-melotti) [inline .flake8:3]:
```suggestion
max-complexity = 10
```

File: cherry_picker/cherry_picker.py [hunk 2]
```diff
@@ -542,7 +553,9 @@ def abort_cherry_pick(self):
         state = self.get_state_and_verify()
         if state != WORKFLOW_STATES.BACKPORT_PAUSED:
             raise ValueError(
-                f"One can only abort a paused process. Current state: {state}. Expected state: {WORKFLOW_STATES.BACKPORT_PAUSED}"
+                "One can only abort a paused process. "
+                f"Current state: {state}. "
+                f"Expected state: {WORKFLOW_STATES.BACKPORT_PAUSED}"
```
Comment (ezio-melotti) [inline cherry_picker/cherry_picker.py:558]:
```suggestion
                f"One can only abort a paused process. "
                f"Current state: {state}. "
                f"Expected state: {WORKFLOW_STATES.BACKPORT_PAUSED}"
```
Any reason not to keep the `f` on all strings?
It seems more consistent and less error prone to add it.  Same applies to other strings below.

Comment (ezio-melotti) [reply cherry_picker/cherry_picker.py:558]: I generally do, but I'm not sure if there are already conventions one way or the other.  The Ruff rule you mentioned makes sense in the context or lone strings without placeholders; in this case I consider the 3 strings literals (that are implicitly concatenated) as part of the same string, and I would add the same prefix(es) to all chunks.

Comment (hugovk) [reply cherry_picker/cherry_picker.py]: Added!

**Comment by hugovk** [comment]
Same comment applies about the `safety check`: https://github.com/python/cherry-picker/pull/101#issuecomment-1872971559.

File: cherry_picker/cherry_picker.py [hunk 3]
```diff
@@ -151,7 +152,9 @@ def set_paused_state(self):
         set_state(WORKFLOW_STATES.BACKPORT_PAUSED)
 
     def remember_previous_branch(self):
-        """Save the current branch into Git config to be able to get back to it later."""
+        """Save the current branch into Git config
+        to be able to get back to it later.
```
Comment (AA-Turner) [inline cherry_picker/cherry_picker.py:156]: This seems odd

File: cherry_picker/cherry_picker.py [hunk 4]
```diff
@@ -359,7 +370,7 @@ def get_updated_commit_message(self, cherry_pick_branch):
         return updated_commit_message
 
     def amend_commit_message(self, cherry_pick_branch):
-        """ prefix the commit message with (X.Y) """
+        """prefix the commit message with (X.Y)"""
```
Comment (AA-Turner) [inline cherry_picker/cherry_picker.py:373]:
```suggestion
        """Prefix the commit message with (X.Y)"""
```

**Review by AA-Turner — APPROVED** [review]

File: cherry_picker/cherry_picker.py (hunk 3 above)
Comment (hugovk) [reply cherry_picker/cherry_picker.py:156]: Would this be better?

```suggestion
        """Save the current branch into Git config to be able to get back to it
        later.
```

Comment (AA-Turner) [reply cherry_picker/cherry_picker.py:156]: I have an aversion to splitting a one line docstring purely for line-length reasons. Perhaps:

```suggestion
        """Save the current branch into Git config, to be used later."""
```

File: cherry_picker/cherry_picker.py [hunk 5]
```diff
@@ -151,7 +152,8 @@ def set_paused_state(self):
         set_state(WORKFLOW_STATES.BACKPORT_PAUSED)
 
     def remember_previous_branch(self):
-        """Save the current branch into Git config to be able to get back to it later."""
+        """Save the current branch into Git config, to be used later."""
+        """
```
Comment (hugovk) [inline cherry_picker/cherry_picker.py:156]:
```suggestion
```

**Pull request merged**